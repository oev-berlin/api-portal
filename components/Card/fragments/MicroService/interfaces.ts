interface MicroServiceProps {
    serviceName: string,
    isMicroservice: boolean,
}

export type { MicroServiceProps };
