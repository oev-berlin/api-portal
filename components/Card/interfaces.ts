import { ProjectData } from '../../utils/interfaces';

interface CardProps{
    projectData: ProjectData
}

export type { CardProps };
