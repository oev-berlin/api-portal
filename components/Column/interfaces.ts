import {projectData} from "../../utils/interfaces";

interface columnData {
    name: string,
    projects: projectData[],
}

export {columnData};