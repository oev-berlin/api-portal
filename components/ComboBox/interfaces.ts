export interface ComboBoxProps{
    searchOptions: string[],
}
