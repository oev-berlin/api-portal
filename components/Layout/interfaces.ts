import React from 'react';

export interface LayoutProps {
    searchOptions: string[],
    children: React.ReactNode,
}
