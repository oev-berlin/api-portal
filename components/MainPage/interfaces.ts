type docType = "backend" | "microservice";

interface projectData {
    id: string,
    name: string,
    description: string,
    swaggerInformation: object,
    docsType: docType,
    microservices: string[],
    externalServices: string[]
}

interface columnData {
    name: string,
    projects: JSON[]
}

export {projectData, columnData};

