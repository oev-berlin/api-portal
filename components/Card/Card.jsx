import React from 'react';
import PropTypes from 'prop-types';
import styles from './Card.module.css';

export const Card = ({ title, imageSrc, imageAlt, ...props }) => (
  <div className={styles.card} {...props}>
    {title.length > 0 && <h3 className={styles.title}>{title}</h3>}
    {imageSrc.length > 0 && (
      <img className={styles.image} src={imageSrc} alt={imageAlt} />
    )}
  </div>
);

Card.propTypes = {
  title: PropTypes.string,
  imageSrc: PropTypes.string,
  imageAlt: PropTypes.string,
};

Card.defaultProps = {
  title: '',
  imageSrc: '',
  imageAlt: '',
};
