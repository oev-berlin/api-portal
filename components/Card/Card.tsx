import {CardActionArea, Grid, Button} from '@mui/material';
import {Microservice} from "./fragments";
import {CardOuterContainer, CardTitle, CardInnerContainer} from "./styles";
import {projectData} from "../../utils/interfaces";

export const Card = ({projectData}: projectData) => (
    <CardOuterContainer>
        <CardActionArea component={Button} sx={{textTransform:"none"}}>
                <CardTitle>
                    {projectData.name}
                </CardTitle>
            <CardInnerContainer container spacing={2}>
                {
                    projectData.microservices.map((microserviceName: string) =>
                        (
                            <Microservice
                                serviceName={microserviceName}
                                key={microserviceName}
                            />
                        )
                    )}
            </CardInnerContainer>
        </CardActionArea>
    </CardOuterContainer>

);