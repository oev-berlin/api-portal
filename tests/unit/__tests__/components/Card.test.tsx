import { render, cleanup } from '@testing-library/react';
import { toHaveNoViolations } from 'jest-axe';
import React from 'react';
import { Card } from '../../../../components/Card';
import { projectData } from '../../../../utils/interfaces';
// eslint-disable-next-line import/extensions
import 'jest-styled-components';
import { createProjectData } from '../../../../utils/testUtilities';

expect.extend(toHaveNoViolations);

const setupComponent = ({ projectData }: { projectData: projectData }) => render(
  <Card projectData={projectData} />,
);

describe('Card', () => {
  afterEach(() => {
    cleanup();
  });

  describe('UI Tests', () => {
    it('Card component (type microservice) should render without crashing', async () => {
      const screen = setupComponent({
        projectData: createProjectData({
          docsType: 'microservice',
        }),
      });
      expect(screen.container).toBeDefined();
    });

    it('Card component (type backend) should render without crashing', async () => {
      const screen = setupComponent({
        projectData: createProjectData({
          docsType: 'backend',
        }),
      });
      expect(screen.container).toBeDefined();
    });

    it('Card component (without microservices) should render without crashing', async () => {
      const screen = setupComponent({
        projectData: createProjectData({
          microservices: [],
        }),
      });
      expect(screen.container).toBeDefined();
    });
    it('Card component (without external services) should render without crashing', async () => {
      const screen = setupComponent({
        projectData: createProjectData({
          externalServices: [],
        }),
      });
      expect(screen.container).toBeDefined();
    });

    it('Card component should render correctly', async () => {
      const screen = setupComponent({
        projectData: createProjectData({}),
      });
      await screen
        .findByText(/Test Project/)
        .then((element) => expect(element).toBeInTheDocument());
      await screen
        .findByText(/PDF Service/)
        .then((element) => expect(element).toBeInTheDocument());
      await screen
        .findByText(/Email Service/)
        .then((element) => expect(element).toBeInTheDocument());
    });
  });

  describe('Snapshots Tests', () => {
    it('should match a basic snapshot (type microservice)', () => {
      const { container } = setupComponent({
        projectData: createProjectData({
          docsType: 'microservice',
        }),
      });
      expect(container).toMatchSnapshot();
    });

    it('should match a basic snapshot (type backend)', () => {
      const { container } = setupComponent({
        projectData: createProjectData({
          docsType: 'backend',
        }),
      });
      expect(container).toMatchSnapshot();
    });

    it('should match a basic snapshot (without microservices)', () => {
      const { container } = setupComponent({
        projectData: createProjectData({
          microservices: [],
        }),
      });
      expect(container).toMatchSnapshot();
    });

    it('should match a basic snapshot (without external services)', () => {
      const { container } = setupComponent({
        projectData: createProjectData({
          externalServices: [],
        }),
      });
      expect(container).toMatchSnapshot();
    });
  });
});
