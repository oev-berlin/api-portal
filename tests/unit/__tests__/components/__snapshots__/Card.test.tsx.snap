// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Snapshots Tests should match a basic snapshot (type backend) 1`] = `
.c3 {
  text-align: center;
  padding: 0.5rem;
}

.c1 {
  text-align: left;
  padding-left: 0.5rem;
  text-transform: none;
}

.c0 {
  margin-bottom: 1rem;
}

.c2 {
  padding: 1rem;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root c0 css-bhp9pd-MuiPaper-root-MuiCard-root"
    href="/project/1"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root MuiCardActionArea-root css-1hu6zam-MuiButtonBase-root-MuiButton-root-MuiButtonBase-root-MuiCardActionArea-root"
      role="button"
      tabindex="0"
      type="button"
    >
      <h3
        class="c1"
      >
        Test Project
      </h3>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 c2 css-mhc70k-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-xp56uf-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 c3 css-1ps6pg7-MuiPaper-root"
          >
            PDF Service
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-xp56uf-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 c3 css-1ps6pg7-MuiPaper-root"
          >
            Email Service
          </div>
        </div>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`Card Snapshots Tests should match a basic snapshot (type microservice) 1`] = `
.c3 {
  text-align: center;
  padding: 0.5rem;
}

.c1 {
  text-align: left;
  padding-left: 0.5rem;
  text-transform: none;
}

.c0 {
  margin-bottom: 1rem;
}

.c2 {
  padding: 1rem;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root c0 css-bhp9pd-MuiPaper-root-MuiCard-root"
    href="/project/1"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root MuiCardActionArea-root css-1hu6zam-MuiButtonBase-root-MuiButton-root-MuiButtonBase-root-MuiCardActionArea-root"
      role="button"
      tabindex="0"
      type="button"
    >
      <h3
        class="c1"
      >
        Test Project
      </h3>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 c2 css-mhc70k-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-xp56uf-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 c3 css-1ps6pg7-MuiPaper-root"
          >
            PDF Service
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-xp56uf-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 c3 css-1ps6pg7-MuiPaper-root"
          >
            Email Service
          </div>
        </div>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`Card Snapshots Tests should match a basic snapshot (without external services) 1`] = `
.c3 {
  text-align: center;
  padding: 0.5rem;
}

.c1 {
  text-align: left;
  padding-left: 0.5rem;
  text-transform: none;
}

.c0 {
  margin-bottom: 1rem;
}

.c2 {
  padding: 1rem;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root c0 css-bhp9pd-MuiPaper-root-MuiCard-root"
    href="/project/1"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root MuiCardActionArea-root css-1hu6zam-MuiButtonBase-root-MuiButton-root-MuiButtonBase-root-MuiCardActionArea-root"
      role="button"
      tabindex="0"
      type="button"
    >
      <h3
        class="c1"
      >
        Test Project
      </h3>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 c2 css-mhc70k-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-xp56uf-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 c3 css-1ps6pg7-MuiPaper-root"
          >
            PDF Service
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-6 MuiGrid-grid-lg-4 css-xp56uf-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 c3 css-1ps6pg7-MuiPaper-root"
          >
            Email Service
          </div>
        </div>
      </div>
      <span
        class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;

exports[`Card Snapshots Tests should match a basic snapshot (without microservices) 1`] = `
.c1 {
  text-align: left;
  padding-left: 0.5rem;
  text-transform: none;
}

.c0 {
  margin-bottom: 1rem;
}

.c2 {
  padding: 1rem;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

<div>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root c0 css-bhp9pd-MuiPaper-root-MuiCard-root"
    href="/project/1"
  >
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root MuiCardActionArea-root css-1hu6zam-MuiButtonBase-root-MuiButton-root-MuiButtonBase-root-MuiCardActionArea-root"
      role="button"
      tabindex="0"
      type="button"
    >
      <h3
        class="c1"
      >
        Test Project
      </h3>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 c2 css-mhc70k-MuiGrid-root"
      />
      <span
        class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
      />
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
